test/powershell/Modules/Microsoft.PowerShell.Utility/Send-MailMessage.Tests.ps1

```diff
@@ -117,16 +166,154 @@ Describe "Send-MailMessage" -Tags CI, RequireSudoOnUnix {
         $mail = Read-Mail
 
         $mail.FromAddress | Should -BeExactly $InputObject.From
-        $mail.ToAddresses | Should -BeExactly $InputObject.To
+        $mail.ToAddresses.Count | Should -BeExactly ($InputObject.To.Count + $InputObject.Cc.Count + $InputObject.Bcc.Count)
+        $mail.ToAddresses | Should -BeIn ([array]$InputObject.To + $InputObject.Cc + $InputObject.Bcc)
 
         $mail.Headers["From"] | Should -BeExactly $InputObject.From
-        $mail.Headers["To"] | Should -BeExactly $InputObject.To
-        $mail.Headers["Reply-To"] | Should -BeExactly $InputObject.ReplyTo
-        If ($InputObject.Subject -ne $null) {
-            $mail.Headers["Subject"] | Should -BeExactly $InputObject.Subject
+        $mail.Headers["To"] | Should -Not -BeNullOrEmpty
+        $mail.Headers["To"].Split(", ") | Should -BeExactly $InputObject.To
+        If($InputObject.Cc)
+        {
+            $mail.Headers["Cc"] | Should -Not -BeNullOrEmpty
+            $mail.Headers["Cc"].Split(", ") | Should -BeExactly $InputObject.Cc
+        }
+        # BCC addresses can't be tested against mail message header, as this information is by design never included in the mail message itself.
+        If($InputObject.ReplyTo)
+        {
+            $mail.Headers["Reply-To"] | Should -BeExactly $InputObject.ReplyTo
         }
+        $mail.Headers["Subject"] | Should -BeExactly $InputObject.Subject
 
         $mail.MessageParts.Count | Should -BeExactly 1
         $mail.MessageParts[0].BodyData | Should -BeExactly $InputObject.Body
     }
 }
+
+Describe "Send-MailMessage Feature Tests" -Tags Feature, RequireSudoOnUnix {
+    BeforeEach {
+        $server = [netDumbster.smtp.SimpleSmtpServer]::Start(25)
```

Comment:
Can we start the server only once in the Describe for performance?

Reply:
@iSazonov Fixed. Thank you!